File: TP01/C/TP0101.c
// 853355 Davi Puddo

#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

// Descobrir se uma cadeia de caraceteres e um palindromo
bool pali (char* s, int n)
{
	bool result = false;
	if (s)
	{
		int i = 0;
		int y = n-1;
		int w = n;

		while (i < w && s[i] == s[y])
		{
			i++;
			y--;
		}
		if (i == w)
		{
			result = true;
		}
	}
	return (result);
}

int main ()
{
	// Definir dados
	char* data = calloc(80, sizeof(char));
	bool result = false;
	
	while ()
	{
		// Ler dados
		scanf ("%s", data);
		getchar();

		// Verificar se e palindromo
		result = pali(data, strlen(data));

		// Mostrar dados
		if (result)
		{
			printf ("SIM\n");
		}
		else
		{
			printf ("NAO\n");
		}
	}
	return (0);
}

File: TP01/Java/TP0101.java
// 853355 Davi Puddo

import java.util.Scanner;

public class TP0101
{
	// Descobrir se uma string e um palindromo
	public static boolean pali (String s)
	{
		boolean result = false;
		if (s != null)
		{
			int i = 0;
			int y = s.length()-1;
			int w = y;
			while (i < w && s.charAt(i) == s.charAt(y))
			{
				y--;
				i++;
			}
			if (i == w)
			{
				result = true;
			}
		}
		return (result);
	}
		
	public static void main (String[] args)
	{
		// Definir dados
		Scanner sc = new Scanner (System.in);
		String data = null;
		boolean result = false;
		boolean stop = false;
		
		while (stop == false)
		{
			// Ler dados
			data = sc.nextLine();
			
			// Verificar existencia de dados
			if (data != null)
			{
				// Verificar se e um palindromo
				result = pali (data);

				// Condicao de parada
				if (data.length() == 3 && data.charAt(0) == 'F' && data.charAt(1) == 'I' && data.charAt(2) == 'M')
				{
					stop = true;
				}
				else
				{
					// Mostrar dados
					if (result == true)
					{
						System.out.print ("SIM\n");
					}
					else
					{
						System.out.print ("NAO\n");
					}
				}
			}
		}
	}
}
